app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  has_many :comments
+  has_many :likes
+  belongs_to :user
+
+  def update_counter(id)
+    user = User.find_by(id: id)
+    user.posts_counter += 1
+    user.save
+  end
```

Comment:
- [OPTIONAL] This implementation works fine but we can refactor it to make the method look more precise:

```suggestion
  def update_counter
    user.increment!(:posts_counter)
  end
```

Ruby-on-Rails is smart enough to look for the user of a particular post and you don't have to provide the id of the user.

### Hint

It would be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private

```ruby
after_save :update_post_counter

  private

  def update_counter
    author.increment!(:posts_counter)
  end
```
```